subconscious/model.py

```diff
@@ -265,8 +265,19 @@ def get_index_key(cls, column_name):
         return all_keys
 
     @classmethod
-    async def all(cls, db, order_by=None):
-        for redis_key in await cls._get_ids_for_all(db, order_by=order_by):
+    async def all(cls, db, order_by=None, limit=None, offset=None):
+        ids_to_iterate = await cls._get_ids_for_all(db, order_by=order_by)
+        if order_by:
```

Comment:
My thinking is what does limit/offset mean without order by? Perhaps you could do limit without order by (effectively random order), but an offset without an order just makes no sense.

How about we keep it a requirement for now but we can always relax it in the future?